docs/designs/2730-visibility.md

```diff
@@ -0,0 +1,407 @@
+
+Unless I've asked for more verbose output, there should not be any low level
+messages for implementation details that do not provide me with useful
+information about the deployment.
+
+#### Failed cluster deployment
+
+From the previous use case details, I should understand the high level steps in
+the process, even if I don't understand what low level things need to happen to
+implement those steps.
+
+When something goes wrong, I need help understanding what caused the deployment
+to fail, with only the bare low level details necessary to help guide me to
+correcting the cause of the failure. I want to be told specifically what the
+problem was, with as much guidance as possible to help me resolve the problem.
+
+#### Ability to refer to deployment details
+
+Whether my cluster deployment was successful or failed, I may have closed my
+terminal that contained the original output from that process. I need a way to
+refer back to that information so I can either look up information about my
+cluster, or find the debugging information that is going to help me resolve the
+problem. I should be able to access a log of the deployment activity that will
+provide all the details of what happened, including both the output from the
+console as well as more verbose details from the process.
+
+As a user, I may not always be able to understand and fix the cause of a
+failure. In this case, I need this captured information so I can provide it to
+support or other community members so they can help me resolve my issue.
+
+#### Functional Requirements
+
+To address these needs, the following requirements must be met.
+
+##### FR1 - Output should be captured to a persistent log file
+
+If the users terminal closes or content scrolls off their buffer, they need to
+be able to get the details from the cluster creation. This is especially
+critical when something goes wrong and someone else needs to take a look at the
+output to help troubleshoot the failure.
+
+* Cluster creation output should be captured to
+  `~/.config/tanzu/tkg/bootstrap-logs/${CLUSTER_NAME}.log`
+* The log should capture everything, regardless of verbosity level set by the
+  end user for console output
+* Additional debug command output, such as tailing CAPI controller logs on
+  failure, should be captured in the bootstrap log
+* To prevent hidden leaked filesystem space consumption, the bootstrap log
+  should be deleted as part of the `management-cluster delete` or `cluster
+  delete` operations
+
+##### FR2 - Log high level steps
+
+The output should be grouped based on the high level steps of the process. So
+rather than the current:
+
+```txt
+Doing something.
+Start creating cluster...
+Saving file.
+Installing xyz
+Waiting for x...
+```
+
+All output is at the same level and there is no differentiation or
+relevance of each log line. The new output should have clear high level steps:
+
+```txt
+☑️  Determining cluster settings
```

Comment:
I think it would be helpful to see "Determining cluster settings (step 1/7)" in the output so the user can get a sense of where they are in the process

Reply:
I do like this, but I wonder if we will always know up front how many steps will be in the process. I think some things might end up being added or removed dynamically based on the requested settings.

Would you be OK if we leave this out of the proposal for now, then as we solidify the implementation if we find this is actually something we can determine, add it as an implementation detail?